CMakeLists.txt

```diff
@@ -20,6 +20,16 @@ if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
   list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules/")
 endif()
 
+if(EXISTS "${CMAKE_SOURCE_DIR}/third_party_release")
+  file(STRINGS "${CMAKE_SOURCE_DIR}/third_party_release" third_party_tags)
+  foreach(third_party ${third_party_tags})
+    string(REGEX REPLACE "^[ ]+" "" third_party ${third_party})
+    string(REGEX MATCH "^[^=]+" third_party_name ${third_party})
+    string(REPLACE "${third_party_name}=" "" third_party_tag ${third_party})
+    set(${third_party_name} "${third_party_tag}")
+  endforeach()
+endif()
+
```

Comment:
Trying to understand the logic here. We iterate the file and extract the name and version from each entry. In the end,  `third_party_name` and `third_party_tag` contain name and version for last entry of file ?

Reply:
No we are storing all the versions not just the last one. e.g. `${nlohmann-json}` has `v3.9.1`, or `${opentelemetry-proto} `has, `v0.11.0`.
In this PR I use ${opentelemetry-proto} to download it from github.
We can use the same technique for all the other dependencies when they are not available in the `third_party` folder.

Reply:
Ah got it, I missed the `set` command.